### PR TITLE
mutate: basic schemata

### PR DESCRIPTION
**Pull request by joakim-brannstrom**
the goal is to get the infrastructure for program schematas up and running. It means that the test phase can use schematas to speed up mutation testing.
The analyze phase will to begin with be a dummy schemata.

### CODE REVIEW

**Comment by joakim-brannstrom** [comment]
This is the basic design of how I think we can integrate mutation schemata.

**Comment by joakim-brannstrom** [comment]
database_schema

File: plugin/mutate/doc/design/overview.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+array of them to try them out in the real world. To get feedback from the users
+in order to guide the future development.
+
+This will of course need to be re-evaluated when the tool and process has
+reached an acceptable maturity. This is further explained in the chapter
+[PoC](#proof-of-concept).
+
+**Priority 2**: add tests to show it continues working. The intention is to
+continuously demonstrate progress to the stakeholders. An additional benefit is
+a reduction of regressions which would greatly annoy our users.
```
Comment (christoffer-nylen) [inline plugin/mutate/doc/design/overview.md]:
```suggestion
a reduction of regressions which would greatly annoy users and tool contributors.
```

File: plugin/mutate/doc/design/overview.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+**Priority 2**: The PoC should be practical to use for source code mutation for
+applications between 1k-10k SLOC.
+
+The PoC thus help to derive and clarify the use cases.
+
+ - what are the problems when scaling mutation testing?
+ - what are the important usability characteristics needed by a mutation tool for every day development?
+ - what mutations are important from a DO-178C perspective?
+ - explore Offutt's and others idea for a scalable architecture for mutation testing.
```
Comment (christoffer-nylen) [inline plugin/mutate/doc/design/overview.md]: Unclear what `Offutt's and others idea` is. Is there a reference?

**Comment by joakim-brannstrom** [comment]
The test phase is completed. Now to implement an analyser that produces a schemata.

**Pull request merged**